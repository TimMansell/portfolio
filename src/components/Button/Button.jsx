import PropTypes from 'prop-types';

import styles from './Button.module.scss';

export const Button = ({ children, href, target, title }) => {
  return (
    <a
      className={styles.btn}
      href={href}
      target={target}
      title={title}
      rel="noopener noreferrer"
    >
      {children}
    </a>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  href: PropTypes.string.isRequired,
  target: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default Button;
