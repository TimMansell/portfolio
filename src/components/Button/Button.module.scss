@import 'utilities/pxtorem';

.btn {
  font-size: pxtorem(14);
  line-height: 1;
  color: var(--colour-4);
  padding: pxtorem(8);
  border-radius: 7px;
  border: 1px solid var(--bdr-secondary);
  transition: all 0.3s;

  &:hover {
    background: var(--colour-1-alt-1);
  }

  & + & {
    margin-left: var(--vr);
  }

  @media (min-width: $bp-xs) {
    font-size: pxtorem(16);
    padding: pxtorem(10);
  }

  @media (min-width: $bp-sm) {
    font-size: pxtorem(18);
  }

  @media (min-width: $bp-xxxl) {
    font-size: pxtorem(24);
    padding: var(--vr-sm);
  }
}
