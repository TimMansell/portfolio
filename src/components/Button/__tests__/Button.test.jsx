import { it, describe, expect } from 'vitest';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Button from '../Button';

const props = {
  href: 'test href',
  target: 'test target',
  title: 'test title',
};

describe('Button', () => {
  it('should render my component', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<Button {...props}>text</Button>);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<Button {...props}>text</Button>).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('should have correct properties', () => {
    const wrapper = shallow(<Button {...props}>text</Button>);

    expect(wrapper.props().href).toBe(props.href);
    expect(wrapper.props().target).toBe(props.target);
    expect(wrapper.props().title).toBe(props.title);
  });
});
