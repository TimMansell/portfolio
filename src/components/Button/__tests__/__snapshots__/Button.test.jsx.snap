// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should match snapshot 1`] = `
<a
  className="btn"
  href="test href"
  rel="noopener noreferrer"
  target="test target"
  title="test title"
>
  text
</a>
`;
